### DISCUSSION TITLE
Como recuperar as enviroment vars no ReactJS em tempo de execução?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1621] [Perguntas e Respostas]
Galera,

no nodejs eu consigo pegar as env vars no process.env.[variável]... E no react, como consigo recuperar?
O que quero é que o pessoal de infra consiga alterar certos comportamentos sem precisar alterar o código ou rebuildar.

Vi aqui que usando o process.env.NOME_DA_VARIÁVEL eu preciso rebuildar para que o app reflita alterações, como vocês fazem?

Muito obrigado, 
abraços!

**Comment by fdaciuk** [comment]
Oi @wilsonneto-dev! Variáveis de ambiente não funcionam no client side. O uso que você faz delas no React é de forma estática: ao fazer o build da sua aplicação, a ferramenta (normalmente webpack), transforma as chamadas de env vars em strings, que ficam "hard coded" no seu código.

A cada alteração de env vars, é preciso fazer um novo build para adicionar essa string no seu script.

Isso vale para qualquer código client side, seja com lib, framework, ou só vanilla JS :)

**Comment by wilsonneto-swe** [comment]
Entendi, então aqui o melhor é passar pro pessoal da infra que quando alterarem as envvars dar um rebuild?

**Comment by fdaciuk** [comment]
Exatamente :)
